Mathlib/CategoryTheory/Sites/Equivalence.lean

```diff
@@ -102,16 +60,28 @@ instance : IsCoverDense e.inverse J where
     replace := Sieve.downward_closed _ this (e.unit.app Y)
     simpa [g] using this
 
+instance : e.functor.IsDenseSubsite J (e.inverse.inducedTopology J) := by
+  have : J = e.functor.inducedTopology (e.inverse.inducedTopology J) := by
+    ext X S
+    show _ ↔ _ ∈ J.sieves _
+    erw [← GrothendieckTopology.pullback_mem_iff_of_isIso (i := e.unit.app X)]
+    congr!; ext Y f; simp
+  nth_rw 1 [this]
+  infer_instance
+
+theorem coverPreserving : CoverPreserving J (e.inverse.inducedTopology J) e.functor :=
+  IsDenseSubsite.coverPreserving _ _ _
+
 /-- A class saying that the equivalence `e` transports the Grothendieck topology `J` to `K`. -/
 class TransportsGrothendieckTopology : Prop where
   /-- `K` is equal to the induced topology. -/
-  eq_inducedTopology : K = (e.locallyCoverDense J).inducedTopology
+  eq_inducedTopology : K = e.inverse.inducedTopology J
```

Comment:
Can this be replaced with the new `IsDenseSubsite` typeclass? @dagurtomas

Reply:
I would think that for an equivalence of categories, the `TransportsGrothendieckTopology` condition should be equivalent to the `functorPushforward_mem_iff` condition from `IsDenseSubsite` (while the first three should be automatic). However, the main goal of the `TransportsGrothendieckTopology` API is to show that the `HasSheafify` assumption can be transported by an equivalence of categories, and for this we cannot use the main result `sheafEquiv` of this new `IsDenseSubsite` API because of the existence of limits assumption. Then, it could be interesting to write an `iff` lemma for `IsDenseSubsite` in the case of an equivalence, and maybe change the definition of `TransportsGrothendieckTopology`, but I do not think that this needs to be done in this PR.

Reply:
Yes the results below needs to stay. I'm just saying that the type class itself could be replaced, and should it?
Not in this PR anyways.

Reply:
I agree that this is probably equivalent, but I haven't worked out the details. 

#13486 has now been adapted and is ready for review.